runtime/src/legacy_memory_manager/cpp/Memory.cpp

```diff
@@ -430,8 +437,70 @@ inline bool isShareable(ContainerHeader* container) {
     return container == nullptr || container->shareable();
 }
 
+MetaObjHeader* createMetaObject(TypeInfo** location);
```

Comment:
Meta objects are likely required for new MM too.

Reply:
Also some of the meta objects machinery is still in common `Memory.h`.
Does anything prevent us from sharing the entire implementation for now?

Reply:
Right, shared some of it.